cmd/rofl/dotenv.go

```diff
@@ -0,0 +1,190 @@
+		raw := lines[i]
+		line := strings.TrimSpace(raw)
+		if line == "" {
+			continue
+		}
+		// Strip BOM on first line if present.
+		if i == 0 && strings.HasPrefix(line, "\uFEFF") {
+			line = strings.TrimSpace(strings.TrimPrefix(line, "\uFEFF"))
+		}
+		// Full-line comment.
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+		// Optional export prefix.
+		if strings.HasPrefix(line, "export ") {
+			line = strings.TrimSpace(line[len("export "):])
+		}
+
+		eq := strings.IndexByte(line, '=')
+		if eq == -1 {
+			fmt.Fprintf(os.Stderr, "warning: skipping malformed line %d (no '='): %s\n", i+1, raw)
+			continue
+		}
+		key := strings.TrimSpace(line[:eq])
+		if key == "" {
+			fmt.Fprintf(os.Stderr, "warning: skipping malformed line %d (empty key)\n", i+1)
+			continue
+		}
+		valuePart := strings.TrimSpace(line[eq+1:])
+
+		var (
+			val string
+			err error
+		)
+
+		// If the value starts quoted and doesn't close on the same line,
+		// keep appending subsequent lines until we find the closing quote.
+		if startsWithQuote(valuePart) {
+			var consumed int
+			val, consumed, err = collectQuotedValue(valuePart, lines, i)
+			if err != nil {
+				return nil, err
+			}
+			// Skip the extra lines we consumed for this value.
+			i += consumed
+		} else {
+			val, err = parseEnvValue(valuePart)
+			if err != nil {
+				return nil, fmt.Errorf("line %d: %w", i+1, err)
+			}
+		}
+
+		res[key] = val
+	}
+	return res, nil
+}
+
+// parseEnvValue parses a single .env value (possibly quoted) and strips trailing comments when unquoted.
+func parseEnvValue(s string) (string, error) {
+	if s == "" {
+		return "", nil
+	}
+	if strings.HasPrefix(s, "\"") {
+		v, err := parseQuotedValue(s, '"', true)
+		return v, err
+	}
+	if strings.HasPrefix(s, "'") {
+		v, err := parseQuotedValue(s, '\'', false)
+		return v, err
+	}
+	// Unquoted value: stop before an unquoted comment start '#'
+	// when it's the first character or preceded by whitespace.
+	for i := 0; i < len(s); i++ {
+		if s[i] == '#' {
+			if i == 0 || isSpace(s[i-1]) {
+				return strings.TrimSpace(s[:i]), nil
+			}
+		}
+	}
+	return strings.TrimSpace(s), nil
+}
+
+// startsWithQuote reports whether the value begins with a single or double quote.
+func startsWithQuote(s string) bool {
+	return len(s) > 0 && (s[0] == '"' || s[0] == '\'')
+}
+
+// collectQuotedValue accumulates a possibly multi-line quoted value until the closing quote.
+// Returns the parsed value, the number of extra lines consumed (beyond the current one), or an error.
+func collectQuotedValue(valuePart string, lines []string, startIdx int) (string, int, error) {
+	delim := valuePart[0]
+	unescape := delim == '"'
+	combined := valuePart
+	consumed := 0
+
+	for {
+		v, e := parseQuotedValue(combined, delim, unescape)
+		if e == nil {
+			return v, consumed, nil
+		}
+
+		// Only continue if this is the unterminated-quote condition.
+		if !strings.HasPrefix(e.Error(), "unterminated quoted value:") {
```

Comment:
Given that you are generating these errors below, just create a new error type using `errors.New` and then match on it using `errors.Is` instead of matching strings.

And I would rewrite the above using a switch statement, e.g.:
```go
switch {
case err == nil:
    return v, consumed, nil
case errors.Is(err, errUnterminatedQuote):
    // ...
default:
    return "", 0, fmt.Errorf("line %d: %w", startIdx+1, err)
}
```